image/rootfs.go

```diff
@@ -38,7 +38,7 @@ func (r *RootFS) Append(id layer.DiffID) {
 func (r *RootFS) Clone() *RootFS {
 	newRoot := NewRootFS()
 	newRoot.Type = r.Type
-	newRoot.DiffIDs = append(r.DiffIDs)
+	newRoot.DiffIDs = append([]layer.DiffID{}, r.DiffIDs...)
```

Comment:
Maybe we can be more explicit here and do

```suggestion
newRoot.DiffIDs = make([]layer.DiffID, len(r.DiffIDs)
copy(newRoot.DiffIDs, r.DiffIDs)
```

Reply:
But in any case this works... or also `newRoot.DiffIDs = append(newRoot.DiffIDs, r.DiffIDs...)`